### PR TITLE
Arc 1997 increase page size to 100

### PR DESCRIPTION
**Pull request by bgvozdev**
**What's in this PR?**
Changing cursor type to be page-aware

**Why**
So we can seamlessly change page sizes

**Added feature flags**
Not yet, should I? Will rollback if something goes wrong 🤠

### CODE REVIEW

File: src/sync/sync-utils.ts [hunk 1]
```diff
@@ -10,6 +10,55 @@ import { SyncType, TaskType } from "~/src/sync/sync.types";
...
+	 * 									the old sync is still in progress with pageNo size equals 20
+	 */
+	constructor(pageNoOrSerialisedCursor: string | number, perPage: number) {
+		if (!pageNoOrSerialisedCursor || Number(pageNoOrSerialisedCursor)) {
+			this.pageNo = Number(pageNoOrSerialisedCursor) || 1;
+			// Historical value we used to have in our code which was used for plain-number cursors.
+			// Ignoring perPage argument for seamless migration
+			this.perPage = 20;
+		} else {
+			const parsed = JSON.parse("" + pageNoOrSerialisedCursor) as PageSizeAwareCounterCursor;
```
Comment (gxueatlassian) [inline src/sync/sync-utils.ts:28]: How about the graphQL cursor, which is just a string?

Comment (bgvozdev) [reply src/sync/sync-utils.ts:28]: This logic is only for counter-based cursor, which is PRs and Builds. The rest entity types (graphQl) should continue working as they are now

Comment (gxueatlassian) [reply src/sync/sync-utils.ts:28]: How about extract this file? I mistakenly thought this new class is also related to sync-utils.

Comment (bgvozdev) [reply src/sync/sync-utils.ts]: Done :)

**Review by gxueatlassian — APPROVED** [review]

**Pull request merged**